File: python/program/Clothing.py
#Saya Nur Ainun 2202046 mengerjakan LATIHAN PRAKTIKUM 2 dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
#untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin. 

from Product import Product


class Clothing(Product):
    __size = ""
    __material = ""
    __gender = ""

    #constructor
    def __init__(self, size="", material="", gender=""):
        self.__size = size
        self.__material = material
        self.__gender = gender

    # Getter and Setter size
    def getSize(self):
        return self.__size
    def setSize(self, size):
        self.__size = size

    # Getter and Setter material
    def getMaterial(self):
        return self.__material
    def setMaterial(self, material):
        self.__material = material

    # Getter and Setter gender
    def getGender(self):
        return self.__gender
    def setGender(self, gender):
        self.__gender = gender

File: c++/Program/Clothing.cpp
/* Saya Nur Ainun 2202046 mengerjakan Latihan Praktikum 2 dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin. */

//import library and file
#include <bits/stdc++.h>
#include "Product.cpp"

class Clothing : public Product
{
    private:
    string size;
    string material;
    string gender;

    public:
    //constructor
    Clothing()
    {
    
    }

    Clothing(string size, string material, string gender){
        this->size = size;
        this->material = material;
        this->gender = gender;
    }

    //Getters and Setters for size
    string getSize()
    {
        return this->size;
    }

    void setSize(string size){
        this->size = size;
    }

    //Getter and Setter for Material
    string getMaterial(){
        return this->material;
    }

    void setMaterial(string material){
        this->material = material;
    }

    //Getters and Setters for gender
    string getGender()
    {
        return this->gender;
    }

    void setGender(string gender)
    {
        this->gender = gender;
    }

    // destructor
    ~Clothing() {}
};

